src/Tasks.UnitTests/RoslynCodeTaskFactory_Tests.cs

```diff
@@ -19,6 +26,70 @@ public class RoslynCodeTaskFactory_Tests
     {
         private const string TaskName = "MyInlineTask";
 
+        [Fact]
+        [SkipOnPlatform(TestPlatforms.AnyUnix, ".NETFramework 4.0 isn't on unix machines.")]
```

Comment:
Did we? All I remember is the comment you had above. I had a series of commits in which I tried to make 5106.csproj build on core; building a project on its own was easy but making it also compile a C# file stymied my progress for a bit. (We don't currently have any other tests that use RunnerUtilities.ExecMSBuild with a project that has a C# file to compile, and if we do it some other way (without relying on framework), I didn't see it.) Ultimately, I think it matters that it works on both core and framework, and this still verifies that it works on both, just not on other platforms.

If it's a dependency built as a normal part of build, I'd worry that we aren't testing anything; if it's already loaded into the current appdomain, the test could pass without this change. (And even if I have a test that currently fails without the change and passes with it, I'd worry that we'd start loading it at some point before this test just by introducing a new unit test, and then we could break this without noticing.

Reply:
> If it's a dependency built as a normal part of build, I'd worry that we aren't testing anything; if it's already loaded into the current appdomain, the test could pass without this change.

That won't be the case if you continue to use ExecMSBuild and push it into its own process, and it's also the sort of thing that you could validate in the test. I think it's worth doing this to avoid OS-specific tests where possible.